kernel/memory/Vmm.cpp

```diff
@@ -8,57 +8,155 @@
 #include <Bitmap.h>
 #include <Memory.h>
 #include <Lazy.h>
+#include <UnitConverter.h>
 
 namespace Npk::Memory
 {
-    VmRange* VMM::AllocStruct()
+    constexpr const size_t VmmSlabSizes[(size_t)VmmMetaType::Count] = 
+    { 
+        sizeof(VmRange), 
+        sizeof(VmHole)
+    };
+
+    constexpr size_t VmmMetaSlabPages = 1;
+
+    VmmMetaSlab* VMM::CreateMetaSlab(VmmMetaType type)
     {
-        sl::ScopedLock scopeLock(allocLock);
-        
-        VmSlabAlloc* allocator = rangesAlloc;
-        while (allocator != nullptr)
+        const size_t index = static_cast<size_t>(type);
+        const size_t size = VmmSlabSizes[index];
+
+        const uintptr_t base = PMM::Global().Alloc(VmmMetaSlabPages) + hhdmBase;
+        const size_t totalSpace = VmmMetaSlabPages * PageSize;
+        const uintptr_t usableBase = sl::AlignUp(base + sizeof(VmmMetaSlab), size);
+        const size_t usableSpace = totalSpace - (usableBase - base);
+
+        size_t slabCount = usableSpace / size;
+        size_t bitmapBytes = sl::AlignUp(slabCount, 8) / 8;
+        while (slabCount * size + bitmapBytes > usableSpace && slabCount > 0)
```

Comment:
i think that operator precedence should be fine, but probably adding some parenthesis could make this more readable? sice there are even 2 ">" operators.